docs/proposal/content-library-support.md

```diff
@@ -0,0 +1,134 @@
+hosts workload clusters, we set up a content library and subscribe to the primary content library, and sync the content library items(templates/OVAs) to it. In this way it is easy to manage templates/OVAs and have confidence in the content of the templates/OVAs.
+
+Using content library as the source of templates has another little benefit that enables using local datastore as destination for the VMs, currently we are limited to shared datastore.
+
+## Motivation
+
+Enhance CAPV to clone VM from content library, to make multi-vCenter template management easier.
+
+### Goals
+
+* Use ContentLibrary items as the VM image template.
+
+### Non-Goals
+
+* Turn content library the only supported template source.
+
+## Proposal
+
+### User Stories
+
+#### Story 1
+
+As an admin, I'd like to manage templates/OVAs across the vCenters that can be consumed by CAPV.
+
+#### Story 2
+
+As an cluster user, I'd like to use content library items(templates/OVAs) as the source of cluster nodes.
+
+### Overall Design
+
+The design focuses on how to clone VMs from content library items. For the content library management itself users can reference vSphere documentation.
+
+### Implementation Details/Notes/Constraints
+
+#### Naming scheme
+
+We use `Template` field of `VirtualMachineCloneSpec` to specify the source we want to clone,
+since we want to augment it to also from the content library, we need to have a way to differentiate the source.
+I propose we use URI scheme for that, it has the format: `scheme://location` which
+
+* scheme, can be vsphere|library
+  * vsphere for conventional vm templates
+  * library for OVA/templates from content library
+* to make it backward compatible, the scheme part can be omitted for conventional vm templates, so if we encounter
+  template without scheme part, we assume it is vsphere template, existing CAPV does not need to do anything.
+
+#### API changes
+
+##### VirtualMachineCloneSpec CR
+
+The `VirtualMachineCloneSpec` has following changes, the spec is used to specify clone spec for `VsphereVM`, it needs the following change to deal with newly added content library capabilities.
+
+```golang
+type VirtualMachineCloneSpec struct {
+  // Template is the name or inventory path of the template used to clone
+  // the virtual machine. It can be a vSphere template, or a content library item. For example:
+  // vsphere template: [vsphere://location of the vsphere template], vsphere:// can be omitted for backward compatibility
+  // content library: [library://name of the content library item]
+  // note that this has impact on cloneMode field, since it is only possible to do full clone from content library items,
+  // and we should have validation rule for this.
+  // +kubebuilder:validation:MinLength=1
+  Template string `json:"template"`
```

Comment:
If we introduce the URI scheme as only valid way, we would have a breaking change.

If we add it as a breaking change, we would have to add it using a new API Version (`v1beta2`) which then also includes converting from v1beta1 to v1beta2 (adding the `vsphere://` prefix).

An alternative to introduce it as a breaking change would be: values not having a prefix will be treated as `vsphere://<value>`.

Reply:
I think this is defined in https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/pull/1839/files#diff-05d6df7ed3848d35aa732cfdeb480d8a394f42809bc3489bc87a9bcdcc0fcc6dR93 (`to make it backward compatible, the scheme part can be omitted for conventional vm templates, so if we encounter template without scheme part, we assume it is vsphere template, existing CAPV does not need to do anything.`)

Reply:
Oh looks like I missed that 😃 thx